codegen/smithy-aws-swift-codegen/src/main/kotlin/software/amazon/smithy/aws/swift/codegen/customization/route53/Route53InvalidBatchErrorIntegration.kt

```diff
@@ -0,0 +1,90 @@
+package software.amazon.smithy.aws.swift.codegen.customization.route53
+
+import software.amazon.smithy.aws.swift.codegen.restxml.AWSRestXMLHttpResponseBindingErrorGenerator
+import software.amazon.smithy.model.Model
+import software.amazon.smithy.model.shapes.ServiceShape
+import software.amazon.smithy.swift.codegen.SwiftDelegator
+import software.amazon.smithy.swift.codegen.SwiftDependency
+import software.amazon.smithy.swift.codegen.SwiftSettings
+import software.amazon.smithy.swift.codegen.SwiftWriter
+import software.amazon.smithy.swift.codegen.core.CodegenContext
+import software.amazon.smithy.swift.codegen.integration.ProtocolGenerator
+import software.amazon.smithy.swift.codegen.integration.SectionWriter
+import software.amazon.smithy.swift.codegen.integration.SectionWriterBinding
+import software.amazon.smithy.swift.codegen.integration.SwiftIntegration
+import software.amazon.smithy.swift.codegen.model.expectShape
+
+class Route53InvalidBatchErrorIntegration : SwiftIntegration {
+    override fun enabledForService(model: Model, settings: SwiftSettings): Boolean {
+        return model.expectShape<ServiceShape>(settings.service).isRoute53
+    }
+
+    override val sectionWriters: List<SectionWriterBinding>
+        get() = listOf(
+            SectionWriterBinding(AWSRestXMLHttpResponseBindingErrorGenerator.RestXMLResponseBindingSectionId, httpResponseBindingErrorGenerator)
+        )
+
+    private val httpResponseBindingErrorGenerator = SectionWriter { writer, previousCode ->
+        val operationErrorName = writer.getContext("operationErrorName") as String
+        if (operationErrorName == "ChangeResourceRecordSetsOutputError") {
+            writer.openBlock("if let customBatchError = CustomInvalidBatchError.makeFromHttpResponse(httpResponse) {", "}") {
+                writer.openBlock("let invalidChangeBatchError = InvalidChangeBatch(", ")") {
+                    writer.write("customError: customBatchError,")
+                    writer.write("headers: httpResponse.headers,")
+                    writer.write("statusCode: httpResponse.statusCode")
+                }
+                writer.write("self = .invalidChangeBatch(invalidChangeBatchError)")
+                writer.write("return")
+            }
+        }
+        writer.write(previousCode)
+    }
+
+    override fun writeAdditionalFiles(ctx: CodegenContext, protocolGenerationContext: ProtocolGenerator.GenerationContext, delegator: SwiftDelegator) {
+        delegator.useFileWriter("${ctx.settings.moduleName}/models/ChangeResourceRecordSetsOutputError+Customization.swift") { writer ->
+            writer.addImport(SwiftDependency.CLIENT_RUNTIME.target)
+            renderCustomInvalidBatchError(writer)
+            renderInvalidChangeBatch(writer)
```

Comment:
It would be awesome to have a way to just write this in swift, since it isn't dependent on the smithy model. 
I filed [this issue](https://github.com/awslabs/aws-sdk-swift/issues/789) to investigate some mechanism that would allow us to write service specific swift code. I imagine the solution here will be related (if not the same) as solving the service specific integration test problem.